cmd/fetchd/cmd/genasiupgrade.go

```diff
@@ -73,16 +98,82 @@ func ASIGenesisUpgradeCmd(defaultNodeHome string) *cobra.Command {
 	return cmd
 }
 
-func ASIGenesisUpgradeReplaceDenomMetadata() {}
+func ASIGenesisUpgradeReplaceDenomMetadata(cdc codec.Codec, appState *map[string]json.RawMessage) error {
+	bankGenState := banktypes.GetGenesisStateFromAppState(cdc, *appState)
+
+	OldBaseDenomUpper := strings.ToUpper(OldBaseDenom)
+	NewBaseDenomUpper := strings.ToUpper(NewBaseDenom)
+
+	denomRegex := getRegex(OldDenom, NewDenom)
+	upperDenomRegex := getRegex(OldBaseDenomUpper, NewBaseDenomUpper)
+	exponentDenomRegex := getPartialRegexLeft(OldBaseDenom, NewBaseDenom)
+
+	for _, metadata := range bankGenState.DenomMetadata {
+		replaceString(&metadata.Base, []*regexPair{denomRegex})
+		if metadata.Name == OldBaseDenomUpper {
+			metadata.Description = NewDescription
+			metadata.Display = NewBaseDenomUpper
+			metadata.Name = NewBaseDenomUpper
+			metadata.Symbol = NewBaseDenomUpper
+		}
+		for _, unit := range metadata.DenomUnits {
+			replaceString(&unit.Denom, []*regexPair{upperDenomRegex})
+			replaceString(&unit.Denom, []*regexPair{exponentDenomRegex})
+		}
+	}
```

Comment:
SImply hardcode the whole DenomMetadata value as literal (list of dict of elements) which would be hardcoded here in this method.